src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs

```diff
@@ -204,6 +207,78 @@ private async Task<DataServiceFeatureMatrix> GetFeatureMatrixInternal( FetchBeha
 			return _FeatureMatrix;
 		}
 
+		/// <summary>
+		/// Limit and sort received measurements if the initial request was split into multiple requests due to uri length.
+		/// </summary>
+		/// <param name="source">Fetched measurements</param>
+		/// <param name="filter">Set filter to check for LimitResult and OrderBy</param>
+		/// <param name="resultSets">Number of requests after splitting the initial request</param>
+		/// <returns>Array of measurements in correct order and size.</returns>
+		private static SimpleMeasurementDto[] LimitAndSortResult( IEnumerable<SimpleMeasurementDto> source, AbstractMeasurementFilterAttributesDto filter, int resultSets )
+		{
+			if( resultSets <= 1 ) return source.ToArray();
+
+			if( filter.OrderBy == null || !filter.OrderBy.Any() )
+				return filter.LimitResult < 0 ? source.ToArray() : source.Take( filter.LimitResult ).ToArray();
+
+			IOrderedEnumerable<SimpleMeasurementDto> orderedResult = null;
+			var firstFilter = filter.OrderBy.First();
+			var sourceArray = source.ToArray();
+
+			//Fast pass for default sorting by time
+			if( filter.OrderBy.Length == 1 && firstFilter.Attribute == WellKnownKeys.Measurement.Time )

```

Comment:
Ich würde diese Weiche eher in die `OrderByAttribute` Methode einbauen. Falls wir zuerst nach einem anderen Attribut sortieren, profitieren wir beim zweiten OrderBy noch von dem einfacheren Zugriff auf `Time`.